Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -90,165 +83,1086 @@ function Sync-AzSOfflineMarketplaceItem {
+    }
+}
+
+function Get-Dependency {
+    param (
+        [parameter(mandatory = $true)]
+        [String] $productid,
 
-        $productid = $_.id
+        [parameter(mandatory = $true)]
+        [String] $resourceGroup
+    )
 
-        # get name of azpkg
-        $uri2 = "$($azureEnvironment.ResourceManagerUrl.ToString().TrimEnd('/'))/subscriptions/$($AzureSubscriptionID.ToString())/resourceGroups/azurestack/providers/Microsoft.AzureStack/registrations/$Registration/products/$($productid)?api-version=2016-01-01"
-        Write-Debug $URI2
-        $Headers = @{ 'authorization' = "Bearer $($Token.AccessToken)"} 
-        $productDetails = Invoke-RestMethod -Method GET -Uri $uri2 -Headers $Headers
-        $azpkgName = $productDetails.properties.galleryItemIdentity
-    
+    $Headers = @{ 'authorization' = "Bearer $($Token.AccessToken)"}
+    $uri = "$($azureEnvironment.ResourceManagerUrl.ToString().TrimEnd('/'))/subscriptions/$($AzureSubscriptionID.ToString())/resourceGroups/$resourceGroup/providers/Microsoft.AzureStack/registrations/$Registration/products/$productid/listDetails?api-version=2016-01-01"
+    $downloadDetails = Invoke-RestMethod -Method POST -Uri $uri -Headers $Headers
 
-        # get download location for apzkg
-        $uri3 = "$($azureEnvironment.ResourceManagerUrl.ToString().TrimEnd('/'))/subscriptions/$($AzureSubscriptionID.ToString())/resourceGroups/azurestack/providers/Microsoft.AzureStack/registrations/$Registration/products/$productid/listDetails?api-version=2016-01-01"
-        $uri3
-        $downloadDetails = Invoke-RestMethod -Method POST -Uri $uri3 -Headers $Headers
-
-        #Create Legal Terms POPUP
-        $a = new-object -comobject wscript.shell
-        $intAnswer = $a.popup($productDetails.properties.description, `
-                0, "Legal Terms", 4)
-        If ($intAnswer -eq 6) {
-           
-            #Output Parameters required for Import
-            $FileExists = Test-Path "$destination\$azpkgName.txt"
-            $DestinationCheck = Test-Path $destination
-            If ($DestinationCheck -eq $false) {
-                new-item -ItemType Directory -force $destination
-            }
-            else {}
+    if ($downloadDetails.properties.dependentProducts)
+    {
+        foreach ($id in $downloadDetails.properties.dependentProducts)
+        {
+            Get-Dependency -productid $id -resourceGroup $resourceGroup
+        }
+    }
 
-            If ($FileExists -eq $true) {Remove-Item "$destination\$azpkgName.txt" -force} else {
-                New-Item "$destination\$azpkgName.txt"
-            }
-            $productDetails.properties|select publisherIdentifier,offer,sku,productKind,vmExtensionType  |out-file "$destination\$azpkgName.txt" -Append
-            $productDetails.properties.productProperties|select version| out-file "$destination\$azpkgName.txt" -Append
+    Write-Host "`nDownloading product: $productid" -ForegroundColor DarkCyan
```

Comment:
This only draws pixels on the screen in the standard PS hosts and cannot be captured in a log file; consider using Write-Verbose

Reply:
Changed most to write-verbose,  want to keep some to display different color.